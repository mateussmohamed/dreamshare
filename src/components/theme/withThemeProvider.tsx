import React, { Component, FC } from 'react'
import { ThemeProvider } from 'styled-components'

import theme from './index'

const getDisplayName = (component: FC) =>
  component.displayName || component.name || 'Component'

const withThemeProvider = (WrappedComponent: FC) =>
  class extends Component {
    static displayName = `withThemeProvider(${getDisplayName(
      WrappedComponent,
    )})`

    render() {
      return (
        <ThemeProvider theme={theme}>
          <WrappedComponent {...this.props} />
        </ThemeProvider>
      )
    }
  }

export default withThemeProvider
