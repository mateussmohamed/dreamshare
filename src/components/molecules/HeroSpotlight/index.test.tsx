import React from 'react'

import { HeroSpotlight } from './index'
import withThemeProvider from '../../theme/withThemeProvider'

describe('[<HeroSpotlight />]', () => {
  const wrap = props => withThemeProvider(() => <HeroSpotlight {...props} />)

  test('matches to snapshot', () => {
    const wrapper = wrap({
      title: 'Hero Title',
      description: 'hero description',
    })

    expect(wrapper).toMatchSnapshot()
  })
})
