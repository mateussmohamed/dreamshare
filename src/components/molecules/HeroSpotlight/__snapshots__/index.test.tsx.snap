// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[<HeroSpotlight />] matches to snapshot 1`] = `[Function]`;
